### PR TITLE
Data processing and UNet for depth estimation

### PR DESCRIPTION
**Pull request by pravinthsam**
data.py: Code to pull data from a S3 bucket, render depthmaps and 2d views from the 3d models and return the respective filenames.
main.py: Code which creates a shapes pytorch dataset and trains a simple U-Net model where input are the 2D views and output are depth maps.

### CODE REVIEW

File: src/data.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import boto3
+import os
+import glob
+import subprocess
+from sklearn.model_selection import train_test_split
+
+def s3_list_subfolders(s3_bucket, s3_path):
+    client = boto3.client('s3')
```
Comment (anithapk) [inline src/data.py:10]: Add a doc-string to all the functions please (https://www.python.org/dev/peps/pep-0008/).

File: src/model.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import torch
+import torch.nn as nn
+
+def double_conv(in_channels, out_channels):
+    return nn.Sequential(
+        nn.Conv2d(in_channels, out_channels, 3, padding=1),
+        nn.ReLU(inplace=True),
```
Comment (anithapk) [inline src/model.py:8]: wondering why u haven't used any batch norm layers?

File: src/train.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        output_image = io.imread(output_path)[:, :, [0]].transpose((2, 0, 1))/255.0
+        output_tensor = torch.from_numpy(output_image).double()
+
+        sample = {'input': input_tensor, 'output': output_tensor}
+
+
+        return sample
+
+def load_dataset():
+    data_transform = transforms.Compose([transforms.ToTensor()])
```
Comment (anithapk) [inline src/train.py:47]: sorry, what does this transform do?

**Review by anithapk — APPROVED** [review]
Looks great! please add docstrings!

File: src/train.py (hunk 3 above)
Comment (pravinthsam) [reply src/train.py:47]: It is used to generate tensors from numpy arrays. I could also add other transforms like randomcrop, randomaffine transforms, etc if I wanted to augment my datasets. [https://pytorch.org/docs/stable/torchvision/transforms.html](https://pytorch.org/docs/stable/torchvision/transforms.html)

File: src/model.py (hunk 2 above)
Comment (pravinthsam) [reply src/model.py:8]: That's a good suggestion. I didn't think about BatchNorm or Dropout since I was trying to overfit on the training data.

File: src/data.py (hunk 1 above)
Comment (pravinthsam) [reply src/data.py:10]: Done.
